File: Data Structures & Algorithms/max-water-container/submission-4.cs
public class Solution {
    public int MaxArea(int[] heights) 
    {
        int l = 0;
        int r = heights.Length - 1;
        int max = 0;
        while(l < r)
        {
            int left = heights[l];
            int right = heights[r];
            int h = Math.Min(left, right);
            int b = r - l;
            max = Math.Max(max, h * b);
            if(left > right)
            {
                r--;
            }
            else
            {
                l++;
            }
        }
        return max;
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-4.java
class Solution {
    public int maxProfit(int[] prices) 
    {
        int maxprofit = 0;
        int l = 0;
        int r = 0;
        while(l <= r && r < prices.length)
        {
            int profit = prices[r] - prices[l];
            maxprofit = Math.max(profit, maxprofit);
            if(prices[l] > prices[r])
            {
                l++;
            }
            else
            {
                r++;
            }
        }
        return maxprofit;
    }
}
